rascal-vscode-extension/src/extension.ts

```diff
@@ -26,463 +26,21 @@
-    const serverOptions: ServerOptions = (deployMode || testDeployMode)
-        ? await buildRascalServerOptions(main)
-        : () => connectToRascalLanguageServerSocket(devPort) // we assume a server is running in debug mode
-            .then((socket) => <StreamInfo> { writer: socket, reader: socket});
-
-    const clientOptions: LanguageClientOptions = {
-        documentSelector: [{ scheme: '*', language: language }],
-    };
-
-    const client = new LanguageClient(language, title, serverOptions, clientOptions, !deployMode);
-    rascalExtensionContext!.subscriptions.push(client);
-
-    await client.start();
-    client.sendNotification("rascal/vfs/register", {
-        port: vfsServer.port
-    });
-    client.onNotification("rascal/showContent", (bp:BrowseParameter) => {
-        showContentPanel(bp.uri);
-    });
-
-    if (!isParametricServer) {
-        client.onNotification("rascal/receiveRegisterLanguage", (lang:LanguageParameter) => {
-            registerLanguage(lang);
-        });
-        client.onNotification("rascal/receiveUnregisterLanguage", (lang:LanguageParameter) => {
-            unregisterLanguage(lang);
-        });
-    }
-
-    let schemesReply:Promise<string[]> = client.sendRequest("rascal/filesystem/schemes");
-
-    schemesReply.then( schemes => {
-        vfsServer.ignoreSchemes(schemes);
-        new RascalFileSystemProvider(client).registerSchemes(schemes);
-    });
-
-
-    return client;
-}
-
-export function deactivate() {
-    if (childProcess) {
-        childProcess.kill('SIGKILL');
-    }
-    vfsServer.dispose();
-}
-
-function registerTerminalCommand() {
-    const command = vscode.commands.registerCommand("rascalmpl.createTerminal", () => {
-        startTerminal(vscode.window.activeTextEditor?.document.uri);
-    });
-    rascalExtensionContext!.subscriptions.push(command);
-}
-
-function registerMainRun() {
-    const command = vscode.commands.registerTextEditorCommand("rascalmpl.runMain", (text, edit, moduleName) => {
-        if (!text.document.uri || !moduleName) {
-            return;
-        }
-        startTerminal(text.document.uri, "--loadModule", moduleName, "--runModule");
-    });
-    rascalExtensionContext!.subscriptions.push(command);
-}
-
-
-function registerImportModule() {
-    const command = vscode.commands.registerTextEditorCommand("rascalmpl.importModule", (text, edit, moduleName) => {
-        if (!text.document.uri || !moduleName) {
-            return;
-        }
-        startTerminal(text.document.uri, "--loadModule", moduleName);
-    });
-    rascalExtensionContext!.subscriptions.push(command);
+    const jars = context.asAbsolutePath(path.join('.', 'assets', 'jars'));
+    const extension = new RascalExtension(context, jars, (deployMode || testDeployMode));
+    context.subscriptions.push(extension);
+    return extension.backwardsCompatibleExternalConsumers();
```

Comment:
I don't understand the meaning of the identifier `backwardsCompatibleExternalConsumers`; "consumers" does not speak to me, and "backward compatible" also not. compatible with what? and why? and is this going to disappear soon then? Perhaps rename or add comments with the declaration of the method to explain what is going on?

Reply:
currently, we enabled other extensions to call "registerLanguage" from the outside. This refactoring prepares for a different way to deploy your own DSL (namely the design discussed in #189) but to remain backwards compatible with existing extension, we still offer the `register` & `unregister` language as exported functionality.

But we should rename this maybe?

Reply:
I would expect that we always have `register` and `unregister` during development time, just that we have different ways of deploying DSLs when we are ready. The way in #189 is most attractive from a branding point of view (we should think about branding Rascal there somehow while still allowing the brand of the DSL to float on top).

Reply:
During develoment time you don't need this, if #189 is finished. Either you are developing your DSL logic inside of 1st level rascal, then you start everything using the REPL.

If you are in second level of your own extension (with a DSL) this setup is really cumbersome (as you need to enable all other extensions to load, just so that your own extension can find rascal). So #189 will also make second-level development of your extension easier.


But this is not the discussion here. The pre-#189 design is that we export 3 functions, such that external extension can call `register` & `unregister` for any language. This function keeps that working, untill we at a later point in time can truly deprecated it, and even later, remove it.

Reply:
ok